marathon/src/main/scala/io/buoyant/marathon/v2/AppIdNamer.scala

```diff
@@ -37,93 +35,99 @@ class AppIdNamer(
    * Marathon master.
    */
   def lookup(path: Path): Activity[NameTree[Name]] =
-    appsActivity.map {
-      case apps =>
+    if (path.isEmpty) Activity.value(NameTree.Neg)
+    else {
+      // each time the map of all Apps updates, find the
+      // shortest-matching part of `path` that exists as an App ID.
+      val possibleIds = (1 to path.size).map(path.take(_))
+      appsActivity.map { apps =>
         Trace.recordBinary("marathon.path", path.show)
-
-        // enumerate all possible span lengths that could be marathon ids
-        val possibleSpans: Seq[PathSpan] = 1 to path.size map { i: Int =>
-          PathSpan(path.take(i), path.drop(i))
-        }
-
-        possibleSpans.find(sp => apps(sp.app.show)) match {
-          case Some(PathSpan(app, residual)) =>
-            val id = prefix ++ app
-
+        possibleIds.collectFirst {
+          case app if apps(app) =>
             Trace.recordBinary("marathon.appId", app.show)
-            Trace.recordBinary("marathon.id", id.show)
-            Trace.recordBinary("marathon.found", app.show)
-
-            val addr = getAndMonitorAddr(app.show)
-            NameTree.Leaf(Name.Bound(addr, id, residual))
-
-          case None =>
-            Trace.recordBinary("marathon.notfound", path.show)
```

Comment:
it's redundant to trace both appId and found.  should probably restore notfound.